manifests/instance.pp

```diff
@@ -0,0 +1,212 @@
+#    resultant haproxy.cfg file.
+#
+# [*defaults_options*]
+#   A hash of all the haproxy defaults options. If you want to specify more
+#    than one option (i.e. multiple timeout or stats options), pass those
+#    options as an array and you will get a line for each of them in the
+#    resultant haproxy.cfg file.
+#
+#[*restart_command*]
+#   Command to use when restarting the on config changes.
+#    Passed directly as the <code>'restart'</code> parameter to the service
+#    resource.  #    Defaults to undef i.e. whatever the service default is.
+#
+#[*custom_fragment*]
+#  Allows arbitrary HAProxy configuration to be passed through to support
+#  additional configuration not available via parameters, or to short-circut
+#  the defined resources such as haproxy::listen when an operater would rather
+#  just write plain configuration. Accepts a string (ie, output from the
+#  template() function). Defaults to undef
+#
+#[*config_file*]
+#  Allows arbitrary config filename to be specified. If this is used,
+#  it is assumed that the directory path to the file exists and has
+#  owner/group/permissions as desired.  If set to undef, the name
+#  will be generated as follows:
+#    If $title is 'haproxy', the operating system default will be used.
+#    Otherwise, /etc/haproxy-$title/haproxy-$title.conf (Linux),
+#    or /usr/local/etc/haproxy-$title/haproxy-$title.conf (FreeBSD)
+#    The parent directory will be created automatically.
+#  Defaults to undef.
+#
+# === Examples
+#
+# A single instance of haproxy with all defaults
+# i.e. emulate Class['haproxy']
+#   package{ 'haproxy': ensure => present }->haproxy::instance { 'haproxy': }->
+#   haproxy::listen { 'puppet00':
+#     collect_exported => false,
+#     ipaddress        => $::ipaddress,
+#     ports            => '8140',
+#   }
+#
+# Multiple instances of haproxy:
+#   package{ 'haproxy': ensure => present }
+#   haproxy::instance { 'group1': }
+#   haproxy::listen { 'puppet00':
+#     instance         => 'group1',
+#     collect_exported => false,
+#     ipaddress        => $::ipaddress,
+#     ports            => '8800',
+#     requires         => Package['haproxy'],
+#   }
+#   haproxy::instance { 'group2': }
+#   haproxy::listen { 'puppet00':
+#     instance         => 'group2',
+#     collect_exported => false,
+#     ipaddress        => $::ipaddress,
+#     ports            => '9900',
+#     requires         => Package['haproxy'],
+#   }
+#
+# Multiple instances of haproxy, one with a custom haproxy package:
+#   package{ 'haproxy': ensure => present }
+#   haproxy::instance { 'group1': }
+#   haproxy::listen { 'puppet00':
+#     instance         => 'group1',
+#     collect_exported => false,
+#     ipaddress        => $::ipaddress,
+#     ports            => '8800',
+#     requires         => Package['haproxy'],
+#   }
+#   package{ 'custom_haproxy': ensure => present }
+#   haproxy::instance { 'group2':
+#     haproxy_package  => 'custom_haproxy',
```

Comment:
I was creating the Service[$instance_name] elsewhere.  I've now included an example of how to do it since everyone will need this too.

I also fixed the package_name issue.